### PR TITLE
Fix RiscV installation/uninstallation

### PR DESCRIPTION
**Pull request by SergioGasquez**
- RiscV Rust target was only installed when installing `ESP32C3`. Now its also installed with `ESP32C2`
- Fix `espup uninstall` when several RiscV targets were installed. See #106  for more details
- RiscV Rust targets were not being deleted during uninstall command, now they are if a RiscV target is installed.

### CODE REVIEW

File: src/toolchain/rust.rs [hunk 1]
```diff
@@ -409,6 +409,22 @@ fn install_rust_nightly(version: &str) -> Result<()> {
     Ok(())
 }
 
+/// Unnstalls the RiscV target.
+pub fn uninstall_riscv_target(nightly_version: &str) -> Result<()> {
+    info!("{} Installing Riscv target", emoji::WRENCH);
```
Comment (jessebraham) [inline src/toolchain/rust.rs:414]:
```suggestion
    info!("{} Uninstalling Riscv target", emoji::WRENCH);
```

Typo here, also we seem to be printing "Deleteing RISC-V target" and then immediately printing "Installing RiscV target".

Additionally, I think we should be consistent with how we display these; __RISC-V__ is the "official" way to write this so maybe we should stick with that.

File: src/toolchain/rust.rs [hunk 2]
```diff
@@ -409,6 +409,22 @@ fn install_rust_nightly(version: &str) -> Result<()> {
...
+    cmd!(
+        "rustup",
+        "target",
+        "remove",
+        "--toolchain",
+        nightly_version,
+        "riscv32imac-unknown-none-elf"
```
Comment (jessebraham) [inline src/toolchain/rust.rs:421]: Are we _only_ installing the `imac` target here? Generally we target `imc`, as `imac` requires emulating atomics which has a performance hit, however both _are_ supported.

File: src/main.rs [hunk 3]
```diff
@@ -189,7 +190,7 @@ fn install(args: InstallOpts) -> Result<()> {
 
     exports.extend(llvm.install()?);
 
-    if targets.contains(&Target::ESP32C3) {
+    if targets.contains(&Target::ESP32C3) || targets.contains(&Target::ESP32C2) {
```
Comment (jessebraham) [inline src/main.rs:193]: Maybe a nicer way of handling this (which could be done in a separate PR) would be to implement some helpers on `Target` to determine the architecture. As we continue to add support for addition devices we're going to keep needing to chain expressions with `||` here.

Maybe if we do something like this:

```rust
impl Target {
    pub fn riscv(&self) -> bool {
        !self.xtensa()
    }

    pub fn xtensa(&self) -> bool {
        matches!(self, Target::Esp32 | Target::Esp32s2 | Target::Esp32s3)
    }
}
```

We can then change this line (and the others like it) to something like this:

```rust
if targets.iter().any(|t| t.riscv()) {
    //
}
```

Just a thought, doesn't really matter right now but will likely clean things up in the future. If you don't like this idea then that is fine of course too 😁

**Review by jessebraham — COMMENTED** [review]
Looks good overall, but left a few comments. Feel free to ignore them 😁

File: src/toolchain/rust.rs (hunk 2 above)
Comment (SergioGasquez) [reply src/toolchain/rust.rs:423]: Just updated it to install/uninstall both `imac` and `imc` targets.

**Pull request merged**